### PR TITLE
Update readme and add more timeout parsing tests

### PR DESCRIPTION
**Pull request by timostamm**
Our readme is a bit out of date - this updates it.

Fixes a small issue in parsing grpc timeout header values.

### CODE REVIEW

File: packages/connect-core/src/protocol-connect/connect-parse-timeout.ts [hunk 1]
```diff
@@ -16,34 +16,20 @@ import { Code } from "../code.js";
...
   value: string | null
 ): number | undefined | ConnectError {
   if (value === null) {
     return undefined;
   }
-
-  const regex = /^\d{0,10}$/;
-  const results = regex.exec(value);
-
+  const results = /^\d{1,10}$/.exec(value);
```
Comment (timostamm) [inline packages/connect-core/src/protocol-connect/connect-parse-timeout.ts:27]: Moves all validation into the regexp.

File: packages/connect-core/src/protocol-grpc/grpc-parse-timeout.spec.ts [hunk 2]
```diff
@@ -18,12 +18,18 @@ import { grpcParseTimeout } from "./grpc-parse-timeout.js";
 describe("grpcParseTimeout()", function () {
   it("should parse proper timeout", () => {
     expect(grpcParseTimeout("1H")).toEqual(3600000);
+    expect(grpcParseTimeout("1M")).toEqual(60000);
```
Comment (timostamm) [inline packages/connect-core/src/protocol-grpc/grpc-parse-timeout.spec.ts:21]: We were failing this one.

File: packages/connect-core/src/protocol-grpc/grpc-parse-timeout.ts [hunk 3]
```diff
@@ -15,78 +15,29 @@
...
-    return new ConnectError(
-      `protocol error: invalid grpc timeout format: ${value}`,
-      Code.InvalidArgument
-    );
-  }
-
-  const regex = /(\d{1,9})([HMSmun])$/gm;
-  const results = regex.exec(value);
-
+  const results = /^(\d{1,8})([HMSmun])$/.exec(value);
```
Comment (timostamm) [inline packages/connect-core/src/protocol-grpc/grpc-parse-timeout.ts:27]: Moves all validation into the regexp.

File: packages/connect-node/src/private/io.ts [hunk 4]
```diff
@@ -161,7 +161,7 @@ export function end(stream: stream.Writable): Promise<void> {
 
 export function write(
   stream: stream.Writable,
-  data: Uint8Array | string
+  data: Uint8Array
```
Comment (timostamm) [inline packages/connect-node/src/private/io.ts:164]: We don't pass a string to this anywhere, and it's unlikely we'll ever need to.

**Review by gilwong00 — APPROVED** [review]
LGTM

File: packages/connect-core/src/protocol-grpc/grpc-parse-timeout.spec.ts (hunk 2 above)
Comment (gilwong00) [reply packages/connect-core/src/protocol-grpc/grpc-parse-timeout.spec.ts:21]: good catch!

File: packages/connect-core/src/protocol-grpc/grpc-parse-timeout.ts [hunk 5]
```diff
@@ -15,78 +15,29 @@
...
-      return timeout / 1000;
-    case TimeoutUnit.Nanosecond:
-      return timeout / 1000 / 1000;
-    default:
-      return new ConnectError(
-        `protocol error: invalid grpc timeout unit: ${unit}`,
-        Code.InvalidArgument
-      );
-  }
+  const v = {
```
Comment (gilwong00) [inline packages/connect-core/src/protocol-grpc/grpc-parse-timeout.ts:34]: Should we use a more descriptive variable name?

**Pull request merged**